AxonDeepSeg/download_model.py

```diff
@@ -1,64 +1,86 @@
-        )
-
+        logger.error("An error occured. The model was not downloaded.")
+        sys.exit(DOWNLOAD_ERROR)
     files_after = list(Path.cwd().iterdir())
 
-    # retrieving unknown model folder names
-    folder_name_SEM_model = Path("default-SEM-model-" + sem_release_version)
-    folder_name_TEM_model = Path("default-TEM-model-" + tem_release_version)
-    folder_name_BF_model = Path("default-BF-model-" + bf_release_version)
-
-    if sem_destination.exists():
-        print('SEM model folder already existed - deleting old one')
-        shutil.rmtree(str(sem_destination))
-
-    if tem_destination.exists():
-       print('TEM model folder already existed - deleting old one')
-       shutil.rmtree(str(tem_destination))      
+    # retrieving unknown model folder name
+    folder_name = list(set(files_after) - set(files_before))[0]
 
-    if bf_destination.exists():
-       print('BF model folder already existed - deleting old one')
-       shutil.rmtree(str(bf_destination))
+    if model_destination.exists():
+        logger.info("Model folder already existed - deleting old one")
+        shutil.rmtree(str(model_destination))
+    
+    shutil.move(folder_name, str(model_destination))
 
-    shutil.move(folder_name_SEM_model.joinpath("model_seg_rat_axon-myelin_sem"), str(sem_destination))
-    shutil.move(folder_name_TEM_model.joinpath("model_seg_mouse_axon-myelin_tem"), str(tem_destination))
-    shutil.move(folder_name_BF_model.joinpath("model_seg_rat_axon-myelin_bf"), str(bf_destination))
+def main(argv=None):
+    ap = argparse.ArgumentParser()
+    ap.add_argument(
+        "-m", "--model-name",
+        required=False,
+        help="Model to download. Default: generalist",
+        default='generalist',
+        type=str,
+    )
+    ap.add_argument(
+        "-t", "--model-type",
+        required=False,
+        help="Model type to download. Default: light",
+        default='light',
+        type=str,
+    )
+    ap.add_argument(
+        "-l", "--list",
+        required=False,
+        help="List all available models for download",
+        default=False,
+        action='store_true',
+    )
+    args = vars(ap.parse_args(argv))
 
-    # remove temporary folders
-    shutil.rmtree(folder_name_SEM_model)
-    shutil.rmtree(folder_name_TEM_model)
-    shutil.rmtree(folder_name_BF_model)
+    if args["list"]:
+        logger.info("Printing available models:")
+        for model in MODELS:
+            logger.info(model)
+            model_details = {
+                "MODEL NAME": MODELS[model]['name'],
+                "NUMBER OF CLASSES": MODELS[model]['n_classes'],
+                "OVERVIEW": MODELS[model]['model-info'],
+                "TRAINING DATA": MODELS[model]['training-data'],
+            }
+            pprint.pprint(model_details)
+        sys.exit(SUCCESS)
+    else:
+        download_model(args["model_name"], args["model_type"])
 
-def main(argv=None):
-    download_model()
+if __name__ == "__main__":
```

Comment:
self: 3 lines not covered by tests

Reply:
Done in [d479b6c](https://github.com/axondeepseg/axondeepseg/pull/800/commits/d479b6cdf7843d4394afb17bad6de2395d94a05e)